src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.PooledCrypto.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Security.Cryptography;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        internal static ArraySegment<byte> RentAsn1StringBytes(IntPtr asn1)
+        {
+            return RentDynamicBuffer((ptr, buf, i) => GetAsn1StringBytes(ptr, buf, i), asn1);
+        }
+
+        private static ArraySegment<byte> RentDynamicBuffer<THandle>(NegativeSizeReadMethod<THandle> method, THandle handle)
```

Comment:
Is anything else using this?  Wondering if it actually needs to be another generic method that operates on a delegate or whether its implementation could just be part of RentAsn1StringBytes.

Reply:
Nothing else is using it currently.  But no analysis was done to see what things using the allocating version would benefit from the renting version; so it seems like leaving it like this would save the effort later.